src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -255,19 +258,35 @@ static void registerBukkitEvents() {
 			if (e.equals(PlayerInteractAtEntityEvent.class) || e.equals(PlayerArmorStandManipulateEvent.class)) {
 				continue; // Ignore, registered above
 			}
-			
-			if (!containsSuperclass((Set) registeredEvents, e)) { // I just love Java's generics
+
+			if (!isEventRegistered(handlerList, priority)) { // Check if event is registered
 				Bukkit.getPluginManager().registerEvent(e, listener, priority, executor, Skript.getInstance());
-				registeredEvents.add(e);
 			}
 		}
 	}
-	
-	public static boolean containsSuperclass(Set<Class<?>> classes, Class<?> c) {
-		if (classes.contains(c))
-			return true;
-		for (Class<?> cl : classes) {
-			if (cl.isAssignableFrom(c))
+
+	@NotNull
+	private static HandlerList getHandlerList(Class<? extends Event> e){
+		HandlerList handlerList = null;
+		try {
+			Method method = Bukkit.getPluginManager().getClass().getDeclaredMethod("getEventListeners", Class.class);
```

Comment:
I think it's better to replicate the behaviour of getEventListeners than to use a reflective call to it.